tools/backups2datalad.py

```diff
@@ -0,0 +1,301 @@
+            local_assets = set(dataset_files(dsdir))
+            local_assets.discard(dsdir / dandiset_metadata_file)
+            for a in assets:
+                log.info("Syncing asset %s", a["path"])
+                gid = a["girder"]["id"]
+                dandi_hash = a.get("sha256")
+                if dandi_hash is None:
+                    log.warning("Asset metadata does not include sha256 hash")
+                mtime = a["modified"]  # type: datetime.datetime
+                bucket_url = self.get_file_bucket_url(gid)
+                dest = dsdir / a["path"].lstrip("/")
+                dest.parent.mkdir(parents=True, exist_ok=True)
+                local_assets.discard(dest)
+                deststr = str(dest.relative_to(dsdir))
+                if not dest.exists():
+                    log.info("Asset not in dataset; will copy")
+                    to_update = True
+                    added += 1
+                elif dandi_hash is not None:
+                    if dandi_hash == get_annex_hash(deststr):
+                        log.info(
+                            "Asset in dataset, and hash shows no modification; will not update"
+                        )
+                        to_update = False
+                    else:
+                        log.info(
+                            "Asset in dataset, and hash shows modification; will update"
+                        )
+                        to_update = True
+                        updated += 1
+                else:
+                    stat = dest.stat()
+                    if (
+                        stat.st_size == a["attrs"]["size"]
+                        and stat.st_mtime == mtime.timestamp()
+                    ):
+                        log.info(
+                            "Asset in dataset, and size & mtime match; will not update"
+                        )
+                        to_update = False
+                    else:
+                        log.info(
+                            "Asset in dataset, and size & mtime do not match; will update"
+                        )
+                        to_update = True
+                        updated += 1
+                if to_update:
+                    src = self.assetstore_path / urlparse(bucket_url).path.lstrip("/")
+                    if src.exists():
+                        try:
+                            self.mklink(src, dest)
+                        except Exception:
+                            if self.ignore_errors:
+                                log.warning("cp command failed; ignoring")
+                                continue
+                            else:
+                                raise
+                        log.info("Adding asset to dataset")
+                        ds.repo.add([deststr])
+                        log.info("Adding URL %s to asset", bucket_url)
+                        ds.repo.add_url_to_file(deststr, bucket_url)
```

Comment:
here -- use `batch=True` mode for `add_url_to_file`?

Reply:
Is just adding `batch=True` sufficient, or do I need to add something else at the end of the batch?

Reply:
yeap.  Upon `.save` (or `.repo.commit`) all batched annexes will be stopped/finished before proceeding (there is a lingering issue which was worked-around in a rush but IIRc only in master, more info https://github.com/datalad/datalad/issues/4998)

Reply:
Done.